webview-ui/src/utilities/state.ts

```diff
@@ -55,51 +76,68 @@ type StateUpdater<TState> = React.Reducer<TState, StateMessage>;
  * Takes a message handler for updating state and creates a reducer that can be used in the
  * `useReducer` function.
  */
-function toStateUpdater<TMsgDef extends MessageDefinition, TState>(handler: StateMessageHandler<TMsgDef, TState>): StateUpdater<TState> {
+function toStateUpdater<TMsgDef extends MessageDefinition, TState>(
+    handler: StateMessageHandler<TMsgDef, TState>,
+): StateUpdater<TState> {
     return (state, msg) => {
         if (msg.command in handler) {
-            return handler[msg.command](state, msg.args);
+            return handler[msg.command](state, msg.args as TMsgDef[string]); // TODO: Check this cast
```

Comment:
This is just a type annotation and won't affect the compiled JavaScript, so there's no urgent need to resolve the TODO, but I want to keep it there as a reminder that there's something not quite right with the types here.